pkg/diff/diff.go

```diff
@@ -623,8 +620,13 @@ func (sc *Syncer) Solve(ctx context.Context, parallelism int, dry bool, isJSONOu
 					return nil, err
 				}
 
-				if err := kong.FillPluginsDefaults(&pluginCopy.Plugin, schema); err != nil {
-					return nil, fmt.Errorf("failed filling plugin defaults: %w", err)
+				var oldPlugin *kong.Plugin
+				if e.OldObj != nil && e.OldObj.(*state.Plugin) != nil {
```

Comment:
```suggestion
				if kongStatePlugin, ok := e.Oldobj.(*state.Plugin); ok{
				    oldPlugin = kongStatePlugin.Plugin
				}
```
Usually we use `a, ok := aa.(<someType>)` to do type conversion and check in go. `ok` is true means that `aa` has the required type.

Reply:
thanks, done